### PR TITLE
For comparison, convert auth example to use react-hook-form

### PR DESCRIPTION
**Pull request by flybayer**
This converts the auth example to use react-hook-form so we can have a comparison between this and react-final-form.

Change the diff mode to side-by-side for better comparison:
https://github.com/blitz-js/blitz/pull/767/files?diff=split

### CODE REVIEW

File: examples/auth/app/auth/components/LoginForm.tsx [hunk 1]
```diff
@@ -1,69 +1,79 @@
...
 import login from "app/auth/mutations/login"
 
 type LoginFormProps = {
   onSuccess?: () => void
 }
 
 export const LoginForm = (props: LoginFormProps) => {
+  const methods = useForm({
+    mode: "onBlur",
+    async resolver(values) {
```
Comment (jay-jlm) [inline examples/auth/app/auth/components/LoginForm.tsx:13]: Alternatively, if you are interested in steering users towards using more idiomatic JS:
```
async resolver(value) {
    return Object.keys(values).reduce((acc, key) => {
        if(requiredFields.includes(key) && !isRequiredValid(_values[key])) acc[key] = "Required"
        return acc;
    }, {})
}
```

File: examples/auth/app/auth/components/LoginForm.tsx [hunk 2]
```diff
@@ -1,69 +1,79 @@
...
-            await login({email: values.email, password: values.password})
-            props.onSuccess && props.onSuccess()
-          } catch (error) {
-            if (error.name === "AuthenticationError") {
-              return {[FORM_ERROR]: "Sorry, those credentials are invalid"}
-            } else {
-              return {
-                [FORM_ERROR]:
-                  "Sorry, we had an unexpected error. Please try again. - " + error.toString(),
+      <FormProvider {...methods}>
```
Comment (jay-jlm) [inline examples/auth/app/auth/components/LoginForm.tsx:29]: I see you are following some of the examples in from the lib's own docs here, but this object does not contain only methods, so it's a bit of a misnomer, ihmo.

File: examples/auth/app/auth/components/LoginForm.tsx [hunk 3]
```diff
@@ -1,69 +1,79 @@
...
+        <form
+          onSubmit={methods.handleSubmit(async (values) => {
+            try {
```
Comment (jay-jlm) [inline examples/auth/app/auth/components/LoginForm.tsx:32]: Extract to a function?

**Comment by jay-jlm** [comment]
So happy with this pull request! 
I'm big fan of react-hook-form.

File: examples/auth/app/auth/components/LoginForm.tsx (hunk 2 above)
Comment (flybayer) [reply examples/auth/app/auth/components/LoginForm.tsx:29]: Yeah I agree

File: examples/auth/app/auth/components/LoginForm.tsx (hunk 1 above)
Comment (flybayer) [reply examples/auth/app/auth/components/LoginForm.tsx:13]: I find `reduce` very hard to read for working with objects :)

Comment (jay-jlm) [reply examples/auth/app/auth/components/LoginForm.tsx:13]: I'd say it's more a case of getting used to it but you're not alone, for sure!

**Comment by flybayer** [comment]
Hook form option added in #917

**Pull request closed without merging**